src/dhcp/nm-dhcp-client.c

```diff
@@ -541,10 +541,7 @@ nm_dhcp_client_start_ip6 (NMDhcpClient *self,
 	nm_assert (!priv->duid);
 	nm_assert (client_id);
 
-	if (enforce_duid == NM_DHCP_DUID_ENFORCE_NEVER)
-		priv->duid = NM_DHCP_CLIENT_GET_CLASS (self)->get_duid (self, TRUE);
-	else if (enforce_duid == NM_DHCP_DUID_ENFORCE_LEASE_FALLBACK)
-		priv->duid = NM_DHCP_CLIENT_GET_CLASS (self)->get_duid (self, FALSE);
+	priv->duid = NM_DHCP_CLIENT_GET_CLASS (self)->get_duid (self, !enforce_duid);
```

Comment:
should be `if (!enforce_duid) ... get_duid (self, TRUE);`

Reply:
oh, right... fixed thanks.
I realized also that we don't use nor need anymore the gboolean option in get_duid: reverting the commit that brought that in.